<html><head>
<meta http-equiv="Content-Style-Type" context="text/css">
<title>mysql:14997</title>
<link rel="stylesheet" type="text/css" href="/css/w3ml.css">
</head>
<body>
<p class="button"><a href="14996.html">[前]</a><a href="14998.html">[次]</a><a href="../list/14901-15000.html">[番号順一覧]</a><a href="../thread/14901-15000.html">[スレッド一覧]</a></p>
<p class="id">
mysql:14997
</p>
<div class="header">
<p>
From: 坂井 宏昭 &lt;坂井 宏昭 &lt;sakai@xxxxxxxxxx&gt;&gt;<br>
Date: Wed, 05 Aug 2009 10:39:01 +0900<br>
Subject: [mysql 14997] JavaアプリからのVPN越しの接続について<br>

</p>
</div>
<div class="body">
<pre>はじめまして。
パワーシステムの坂井と申します。

現在、以下の環境にてアプリケーションを開発しております。

OS:WindowsXP
JDK:6u12
DB:MySQL 5.1.34
他:Swing + hibernate

お客様の環境の都合上、テーブルをVPN(NTT FG-100)越しに参照したい
と考えております。
「MySQL Query Browser」で対象のDBにアクセスしデータを参照する
ことは可能なのですが、アプリケーションから接続しようとすると、
以下のようなエラーが発生します。
なお、VPNを介さない場合はアプリケーションからも接続できております。

org.hibernate.exception.JDBCConnectionException: Cannot open connection
    at org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:74)
    at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:43)
    at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:29)
    at org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:426)
    at org.hibernate.jdbc.ConnectionManager.getConnection(ConnectionManager.java:144)
    at org.hibernate.jdbc.JDBCContext.connection(JDBCContext.java:119)
    at org.hibernate.transaction.JDBCTransaction.begin(JDBCTransaction.java:57)
    at org.hibernate.impl.SessionImpl.beginTransaction(SessionImpl.java:1326)
    at newApp.menu.StartPage.&lt;init&gt;(StartPage.java:69)
    at newApp.NewApp.startup(NewApp.java:26)
    at org.jdesktop.application.Application$1.run(Application.java:171)
    at java.awt.event.InvocationEvent.dispatch(Unknown Source)
    at java.awt.EventQueue.dispatchEvent(Unknown Source)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
    at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
    at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
    at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
    at java.awt.EventDispatchThread.run(Unknown Source)
Caused by: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

Last packet sent to the server was 0 ms ago.
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
    at java.lang.reflect.Constructor.newInstance(Unknown Source)
    at com.mysql.jdbc.Util.handleNewInstance(Util.java:406)
    at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:1074)
    at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2103)
    at com.mysql.jdbc.ConnectionImpl.&lt;init&gt;(ConnectionImpl.java:718)
    at com.mysql.jdbc.JDBC4Connection.&lt;init&gt;(JDBC4Connection.java:46)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
    at java.lang.reflect.Constructor.newInstance(Unknown Source)
    at com.mysql.jdbc.Util.handleNewInstance(Util.java:406)
    at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:302)
    at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:282)
    at java.sql.DriverManager.getConnection(Unknown Source)
    at java.sql.DriverManager.getConnection(Unknown Source)
    at org.hibernate.connection.DriverManagerConnectionProvider.getConnection(DriverManagerConnectionProvider.java:110)
    at org.hibernate.jdbc.ConnectionManager.openConnection(ConnectionManager.java:423)
    ... 15 more
Caused by: java.net.SocketException: Can't connect to SOCKS proxy:No route to host: connect
    at java.net.SocksSocketImpl.connect(Unknown Source)
    at java.net.Socket.connect(Unknown Source)
    at java.net.Socket.connect(Unknown Source)
    at java.net.Socket.&lt;init&gt;(Unknown Source)
    at java.net.Socket.&lt;init&gt;(Unknown Source)
    at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:253)
    at com.mysql.jdbc.MysqlIO.&lt;init&gt;(MysqlIO.java:280)
    at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2026)
    ... 28 more

環境の問題かとも思うのですが、「MySQL Query Browser」ではアクセス
できているため、？？？という状況です。
Java系の掲示板でも同様の質問をさせていただいたのですが、特殊なこと
なのか、回答はいただけておりません。

未熟者の質問で申し訳ありませんが、何らかの解決案がありましたら、
是非ご教授願います。

以上、よろしくお願い致します。

=============================================
パワーシステム株式会社
システム開発部　システムソリューション２Ｇ
坂井 宏昭
=============================================

</pre>
</div>
<p class="button"><a href="14996.html">[前]</a><a href="14998.html">[次]</a><a href="../list/14901-15000.html">[番号順一覧]</a><a href="../thread/14901-15000.html">[スレッド一覧]</a></p>
<pre class="list"><span class="current">-&gt;&nbsp;<a href="14997.html">  14997 2009-08-05 10:39 [坂井 宏昭 &lt;sakai@xxx] JavaアプリからのVPN越しの接続について   </a></span>
<span>&nbsp;&nbsp;&nbsp;<a href="14998.html">  14998 2009-08-14 19:00 ┗[舘山 聖司 &lt;tateyan@x]                                       </a></span>
<span>&nbsp;&nbsp;&nbsp;<a href="14999.html">  14999 2009-08-14 19:22 　┗[坂井 宏昭 &lt;sakai@xxx]                                     </a></span>
</pre>


</body></html>